packages/onboarding/src/step-container-v2/components/StepContainerV2/StepContainerV2.tsx

```diff
@@ -50,15 +50,51 @@ export const StepContainerV2 = ( {
 		[ isMediumViewport, isLargeViewport ]
 	);
 
+	const topBarRef = useRef< HTMLDivElement >( null );
+	const [ topBarHeight, setTopBarHeight ] = useState( 0 );
+
+	const stickyBottomBarRef = useRef< HTMLDivElement >( null );
+	const [ stickyBottomBarHeight, setStickyBottomBarHeight ] = useState( 0 );
+
+	useLayoutEffect( () => {
+		const observer = new ResizeObserver( ( entries ) => {
+			entries.forEach( ( entry ) => {
+				const element = entry.target;
+
+				if ( element === topBarRef.current ) {
+					setTopBarHeight( entry.contentRect.height );
+				} else if ( element === stickyBottomBarRef.current ) {
+					setStickyBottomBarHeight( entry.contentRect.height );
+				}
+			} );
+		} );
+
+		if ( topBarRef.current ) {
+			observer.observe( topBarRef.current );
+		}
+		if ( stickyBottomBarRef.current ) {
+			observer.observe( stickyBottomBarRef.current );
+		}
+
+		return () => {
+			observer.disconnect();
+		};
+	}, [] );
+
 	return (
 		<StepContainerV2InternalProvider value={ stepContainerContextValue }>
 			<div
 				className={ clsx( 'step-container-v2', {
 					'medium-viewport': isMediumViewport,
 					'large-viewport': isLargeViewport,
 				} ) }
+				style={ {
+					// @ts-expect-error -- These are valid CSS variables.
+					'--step-container-v2-top-bar-height': `${ topBarHeight ?? 0 }px`,
+					'--step-container-v2-sticky-bottom-bar-height': `${ stickyBottomBarHeight ?? 0 }px`,
```

Comment:
What does the `?? 0` do? Do you want it to default to zero before the layout is measured? Maybe you want to change the `useState` to `useState<null|number>(null)`?